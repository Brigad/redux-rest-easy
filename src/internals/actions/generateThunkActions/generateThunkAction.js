const generateThunkAction = (actionType, resourceName) => normalizedURL => ({
  type: actionType,
  url: normalizedURL,
  resourceName,
});

export default generateThunkAction;
