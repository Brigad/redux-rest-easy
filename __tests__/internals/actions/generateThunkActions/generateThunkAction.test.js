import generateThunkAction from '../../../../src/internals/actions/generateThunkActions/generateThunkAction';

const ACTION_TYPE = '@@rest-easy/fruits/eat/REQUEST';
const RESOURCE_NAME = 'fruits';

describe('generateThunkAction', () => {
  test('no args', () => {
    expect(generateThunkAction()()).toMatchSnapshot();
  });

  test('no resourceName', () => {
    expect(generateThunkAction(ACTION_TYPE)()).toMatchSnapshot();
  });

  test('happy path', () => {
    expect(generateThunkAction(ACTION_TYPE, RESOURCE_NAME)()).toMatchSnapshot();
  });
});
