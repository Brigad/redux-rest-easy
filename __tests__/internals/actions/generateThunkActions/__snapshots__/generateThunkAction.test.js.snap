// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateThunkAction happy path 1`] = `
Object {
  "resourceName": "fruits",
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": undefined,
}
`;

exports[`generateThunkAction no args 1`] = `
Object {
  "resourceName": undefined,
  "type": undefined,
  "url": undefined,
}
`;

exports[`generateThunkAction no resourceName 1`] = `
Object {
  "resourceName": undefined,
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": undefined,
}
`;
